### PR TITLE
[INTERNAL] Refactor middlewareRepository to include specVersion

### PR DESCRIPTION
**Pull request by RandomByte**
Pass new middlewareUtil instance to custom middleware defining spec
version 2.0 or higher

Also:
- middleare is only required on-demand now
- Lazy index.js requires

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,23 +2,48 @@
...
 	// Legacy middleware export. Still private.
 	middleware: {
-		csp: require("./lib/middleware/csp"),
-		discovery: mapLegacyMiddlewareArguments(require("./lib/middleware/discovery")),
-		nonReadRequests: mapLegacyMiddlewareArguments(require("./lib/middleware/discovery")),
-		serveIndex: mapLegacyMiddlewareArguments(require("./lib/middleware/serveIndex")),
-		serveResources: mapLegacyMiddlewareArguments(require("./lib/middleware/serveResources")),
-		serveThemes: mapLegacyMiddlewareArguments(require("./lib/middleware/serveThemes")),
-		versionInfo: mapLegacyMiddlewareArguments(require("./lib/middleware/versionInfo")),
+		csp: "./lib/middleware/csp",
```
[the remark this answers is not in the document]
Comment (RandomByte) [reply index.js]: Drop the export of the middlewares or the legacy mapping?

[the remark this answers is not in the document]
Comment (matz3) [reply index.js]: Both 😄 The middlewareRepository should be used instead, right?

[the remark this answers is not in the document]
Comment (RandomByte) [reply index.js]: Right. But I think we can't do the same for the tasks and processors in UI5 Builder? At least we use them ourselves in various places...

[the remark this answers is not in the document]
Comment (RandomByte) [reply index.js]: Done in latest commit 👍

**Pull request merged**